### PR TITLE
Restrict AutoUpdateVersion to be created/updated for cloud

### PR DESCRIPTION
**Pull request by vapopov**
In this PR added validation to restrict modify `AutoUpdateVersion` for cloud users

Related: https://github.com/gravitational/teleport/blob/c906170e1dc2c7f32a4902c4ba81db2e29efb654/rfd/0144-client-tools-updates.md?plain=1#L216-L224

### CODE REVIEW

File: lib/modules/modules.go [hunk 1]
```diff
@@ -331,7 +332,7 @@ func GetModules() Modules {
 var ErrCannotDisableSecondFactor = errors.New("cannot disable multi-factor authentication")
 
 // ValidateResource performs additional resource checks.
-func ValidateResource(res types.Resource) error {
+func ValidateResource(res any) error {
```
Comment (zmb3) [inline lib/modules/modules.go:335]: Is there any reason that `AutoUpdateVersion` can't implement `types.Resource`?

**Comment by hugoShaka** [comment]
If I understand correctly, this will render any version invalida on cloud.
This means cloud themselves will not be able to set version using tctl and local admin.

File: lib/modules/modules.go (hunk 1 above)
Comment (vapopov) [reply lib/modules/modules.go:335]: new resources implements `types.Resource153` they not compatible with each other

Comment (vapopov) [reply lib/modules/modules.go:335]: agree, changing this one

File: lib/auth/autoupdate/autoupdatev1/service.go [hunk 2]
```diff
@@ -292,6 +293,11 @@ func (s *Service) CreateAutoUpdateVersion(ctx context.Context, req *autoupdate.C
 		return nil, trace.Wrap(err)
 	}
 
+	if modules.GetModules().Features().Cloud && !isAdmin(authCtx) {
+		return nil, trace.AccessDenied("only role %q is allowed to modifying %q in cloud",
```
Comment (hugoShaka) [inline lib/auth/autoupdate/autoupdatev1/service.go:297]: This error is not very actionable for users. I think we should have something like:
> This Teleport instance is running on Teleport Cloud. The "autoupdate_version" resource is managed by the Teleport Cloud team. You can use the "autoupdate_config" resource to opt-in, opt-out or configure update schedules.

File: lib/services/presets.go [hunk 3]
```diff
@@ -184,6 +184,9 @@ func NewPresetEditorRole() types.Role {
 					types.NewRule(types.KindUserTask, RW()),
 					types.NewRule(types.KindIdentityCenter, RW()),
 					types.NewRule(types.KindContact, RW()),
+					types.NewRule(types.KindAutoUpdateVersion, RW()),
+					types.NewRule(types.KindAutoUpdateConfig, RW()),
+					types.NewRule(types.KindAutoUpdateAgentRollout, RW()),
```
[the remark this answers is not in the document]
Comment (hugoShaka) [reply lib/services/presets.go:197]: Hmm, I'm not sure how I'll handle the rollout RPC authorization. I'll work on this soon. I'd rather avoid putting the resource in the preset editor role for now as this resource is not owned by the user.

**Review by hugoShaka — APPROVED** [review]

**Comment by vapopov** [comment]
would appreciate one more review

**Review by sclevine — APPROVED** [review]

**Review by tigrato — APPROVED** [review]

**Pull request merged**